controllers/configurationpolicy_controller.go

```diff
@@ -335,15 +335,111 @@ func getTemplateConfigErrorMsg(err error) string {
 	return fmt.Sprintf("An unexpected error occurred when configuring the template resolver: %v", err)
 }
 
+type objectTemplateDetails struct {
+	kind         string
+	name         string
+	namespace    string
+	isNamespaced bool
+}
+
+func addDetails(templateObj *objectTemplateDetails, unstruct unstructured.Unstructured) {
+	if md, ok := unstruct.Object["metadata"]; ok {
+		//nolint:forcetypeassert
+		metadata := md.(map[string]interface{})
+		if objectName, ok := metadata["name"]; ok {
+			templateObj.name = strings.TrimSpace(objectName.(string))
+		}
+		// override the namespace if specified in objectTemplates
+		if objectns, ok := metadata["namespace"]; ok {
+			templateObj.namespace = strings.TrimSpace(objectns.(string))
+		}
+	}
+
+	if objKind, ok := unstruct.Object["kind"]; ok {
+		templateObj.kind = strings.TrimSpace(objKind.(string))
+	}
+}
```

Comment:
That namespaced bit is annoying - I guess that's the reason this mutates an input, as opposed to constructing the `objectTemplateDetails` based on the `unstruct`?

Is it necessary to be so careful with the accessors and TrimSpace? Does `unstruct.GetName()` not quite work for this?

Reply:
The issue is that the values could be templated, so `addDetails` is used to figure out if anything is in those fields to determine whether namespaces need to be fetched. And, later on after templates have been resolved, it's called again to update the values with the resolved values. Does that answer the first question?

For the second question, this was a copy/paste of existing code. I agree--`TrimSpace` probably isn't necessary, and I'll remove it. It'd be much cleaner to use `GetName()`/`GetNamespace()`. There might be a very minor performance improvement by not using `GetName()` since we're fetching two things from `metadata`, but it'd be much cleaner using the built-in functions.

Reply:
Are you thinking this could look something like this?

```golang
	templateObj.kind = unstruct.GetKind()
	templateObj.name = unstruct.GetName()
	templateObj.namespace = unstruct.GetNamespace()
```

It is much cleaner. And the fact that we're not just passing around the `unstruct` instead is probably a future refactor that could be done, also? (And get rid of `addDetails()` entirely...)

Reply:
Actually, that is also more maintainable too (though not that Kind/Name/Namespace would change much). I'll go ahead and update it.

Reply:
Latest changes have removed the `addDetails()` function in favor of calling `unstruct` functions directly.

Reply:
Another followup to using `unstruct` functions directly. Removing the `strings.TrimSpace()` caused a failure because one of the templates was written with a multi-line value and the `\n` was included in the string, so it turns out we do need to have it.